src/main/java/com/synopsys/integration/alert/channel/msteams/MsTeamsMessage.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * blackduck-alert
+ *
+ * Copyright (c) 2019 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.channel.msteams;
+
+import com.synopsys.integration.alert.common.message.model.ComponentItem;
+import com.synopsys.integration.alert.common.message.model.LinkableItem;
+import com.synopsys.integration.alert.common.message.model.ProviderMessageContent;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class MsTeamsMessage {
+    private Set<LinkableItem> providers = new HashSet<>();
+    private List<MsTeamsSection> sections = new ArrayList<>();
+
+    public void addContent(ProviderMessageContent providerMessageContent) {
+        providers.add(providerMessageContent.getProvider());
+
+        MsTeamsSection msTeamsSection = new MsTeamsSection();
+        msTeamsSection.setProvider(providerMessageContent.getProvider().getValue());
+        msTeamsSection.setTopic(providerMessageContent.getTopic().getValue());
+        providerMessageContent.getSubTopic()
+                .map(LinkableItem::getValue)
+                .ifPresent(msTeamsSection::setSubTopic);
+        sections.add(msTeamsSection);
+
+        List<MsTeamsComponent> components = new ArrayList<>(providerMessageContent.getComponentItems().size());
+        for (ComponentItem componentItem : providerMessageContent.getComponentItems()) {
+            MsTeamsComponent msTeamsComponent = new MsTeamsComponent();
+            msTeamsComponent.setCategory(componentItem.getCategory());
+            msTeamsComponent.setOperation(componentItem.getOperation().toString());
+
+            String componentText = componentItem.getComponent().getValue();
+            if (componentItem.getSubComponent().isPresent()) {
```

Comment:
Rather than having an if-statement, you could potentially do something like this:
```java
1. StringBuilder componentTextBuilder = new StringBuilder(componentItem.getComponent().getValue());
2. componentItem.getSubComponent().map(item -> "/" + item.getValue()).ifPresent(componentTextBuilder::add);
3. msTeamsComponent.setText(componentTextBuilder.toString());
```
I tend to be wary of lines like this: `componentItem.getSubComponent().get().getValue()`. We will have to go back later to fix it for SonarCloud. Using ifPresent with Optionals tends to be a better way to avoid bugs. If you have readability concerns, you can put the fluent code on new lines (our convention is all or nothing with this).